fl4health/preprocessing/autoencoders/loss.py

```diff
@@ -0,0 +1,79 @@
+from typing import Tuple
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+
+class VaeLoss(_Loss):
+    """Implements the loss function used for training CVAEs and VAEs.
+    This loss computes the base_loss (defined by the user) between the input and generated output.
+    It then adds the KL divergence between the estimated distribution (represented by mu and logvar)
+    and the standard normal distribution.
+    """
+
+    def __init__(
+        self,
+        latent_dim: int,
+        base_loss: _Loss = torch.nn.BCELoss(reduction="sum"),
```

Comment:
I think the `base_loss` here is important enough that we shouldn't have a default? That is, we should probably force the user to define it every time?